runtime/dart_isolate.cc

```diff
@@ -736,38 +781,58 @@ DartIsolate::CreateDartVMAndEmbedderObjectPair(
     return {nullptr, {}};
   }
 
-  if (!(*embedder_isolate)->Initialize(isolate, is_root_isolate)) {
-    *error = strdup("Embedder could not initialize the Dart isolate.");
-    FML_DLOG(ERROR) << *error;
+  if (!InitializeIsolate(*embedder_isolate, isolate, is_root_isolate, error)) {
     return {nullptr, {}};
   }
 
-  if (!(*embedder_isolate)->LoadLibraries(is_root_isolate)) {
-    *error =
-        strdup("Embedder could not load libraries in the new Dart isolate.");
+  auto* isolate_data = static_cast<std::shared_ptr<DartIsolate>*>(
+      Dart_IsolateGroupData(isolate));
+  if (isolate_data->get() != embedder_isolate->get()) {
+    *error = strdup("Unexpected isolate data during initialization.");
```

Comment:
Can this ever happen? Should this be UNREACHABLE()?

Reply:
I don't think that should normally happen. I'm not sure flutter engine folks use UNREACHABLE - from what I can tell normal pattern is to use FML_DLOG(ERROR) and exit current function.

@chinmaygarde

Reply:
I wasn't too sure so I probably added a few checks that are too paranoid. If you feel this is unnecessary, please get rid of it (or add a `DCHECK` in its place to document that we thought of this). The engine doesn't use `UNREACHABLE`.

Reply:
Switched to `FML_DCHECK`